### PR TITLE
Issue #17884 Fix 0 values in console table

### PR DESCRIPTION
**Pull request by mikehaertl**
| Q             | A
| ------------- | ---
| Is bugfix?    | ✔️
| New feature?  | ❌
| Breaks BC?    | (unsure)
| Tests pass?   | ✔️
| Fixed issues  | #17884

### CODE REVIEW

**Review by samdark — APPROVED** [review]
👍 Please add a line for changelog.

File: framework/console/widgets/Table.php [hunk 1]
```diff
@@ -133,7 +133,7 @@ public function setRows(array $rows)
     {
         $this->rows = array_map(function($row) {
             return array_map(function($value) {
-                return $value ?: ' ';
+                return $value || $value === 0 || $value === '0' ? $value : ' ';
```
Comment (rob006) [inline framework/console/widgets/Table.php]: Wouldn't be better to use `strlen($value) > 0`? Current condition does not support `0.0`, and there may be even more edge cases.

Comment (mikehaertl) [reply framework/console/widgets/Table.php]: It's one of these decade old weirdnesses of PHP that we don't have a simple way to differentiate `0`,  `0.00`, `0.000` etc. from other "falsey" values like `false`, `null`, `[]` and `''`. 

Given that `$value` can also be an array here, `strlen()` alone is not enough. I'll update like follows (better suggestion welcome):

```php
return $value || $value !== [] && (string) $value !== '' ? $value : ' ';
```

I also considered this alternative but both are somewhat ugly:

```php
return $value || !is_array($value) && strlen($value) !==0 ? $value : ' ';
```

**Comment by mikehaertl** [comment]
@samdark Tests failing now at some other unrelated place. Could you restart them? I may have to tweak the expected string though (no time to set up tests locally, sorry)

File: framework/console/widgets/Table.php (hunk 1 above)
Comment (rob006) [reply framework/console/widgets/Table.php]: What is the purpose of this line in the first place? Why empty cells should be normalized to one space?

Comment (mikehaertl) [reply framework/console/widgets/Table.php]: I didn't look why this is required - but that is what it did before, so I didn't change it. Probably the table would not rendered correctly if there isn't at least 1 single space character even if the cell is empty.

Comment (rob006) [reply framework/console/widgets/Table.php]: Right now `$value || ` condition seems to be redundant.

Comment (mikehaertl) [reply framework/console/widgets/Table.php]: Final version:

```php
return empty($value) && !is_numeric($value) ? ' ' : $value;
```

**Comment by samdark** [comment]
Thank you!

**Pull request merged**